csrc/runtime/executor.cpp

```diff
@@ -208,6 +208,15 @@ void KernelExecutor::compile(
     return;
   }
 
+  const bool has_any_cuda_tensor = std::any_of(
+      fusion->inputs().begin(), fusion->inputs().end(), [](Val* inp) {
+        return inp->isA<TensorView>() && !inp->as<TensorView>()->isCpuScalar();
```

Comment:
Unfortunately I think we'll get some false positives here when we use factory functions. What about this fusion?
```python
  tv0 = fd.ops.full([5], dtype=DataType.Float)
  tv1 = fd.ops.neg(tv0)
  fd.add_output(tv1)
```
This fusion has no inputs at all but produces a CUDA tensor. Instead, we should probably propagate `isCpuScalar()` through expressions, and check that there is at least one non-cpuscalar tensor in the fusion in this check.

Reply:
You're right.
There seem to some other conditions too, let me revise the check.

What do you mean by `propagate isCpuScalar() through expressions`?

Reply:
I mean if we have two cpu scalar tensors and we multiply them, for example, then we get another cpu scalar tensor. Same if we multiply by a non-tensor scalar. But multiplying a CPU scalar tensor by a GPU tensor produces a GPU tensor. I think if we propagate these rules properly than we should only be compiling kernels whenever there is a GPU tensor output.